src/Symfony/Component/Serializer/Extractor/ObjectPropertyListExtractor.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Extractor;
+
+use Symfony\Component\PropertyInfo\PropertyListExtractorInterface;
+
+/**
+ * @author David Maicher <mail@dmaicher.de>
+ */
+class ObjectPropertyListExtractor implements ObjectPropertyListExtractorInterface
+{
+    private $propertyListExtractor;
+    private $objectClassResolver;
+
+    public function __construct(PropertyListExtractorInterface $propertyListExtractor, callable $objectClassResolver)
```

Comment:
Should `$objectClassResolver` be nullable and we should use a fallback to `get_class` like in the current `AbstractObjectNormalizer`?

Reply:
Yes we should, as 90% usage is to use get_class, so we should prefer this usage for DX IMO